test/libponyc-run/ffi-different-returns/main.pony

```diff
@@ -0,0 +1,12 @@
+use "lib:ffi-different-returns-additional"
+
+use @ffi_function[I32]()
+use @pony_exitcode[None](code: I32)
+
+actor Main
+  new create(env: Env) =>
+    let res: (I32 | U8) = @ffi_function[U8]()
```

Comment:
This actually needs to be disallowed - the differing return type needs to be ABI-compatible with all other invocations of the same function. Specifically, I think we need to enforce that all of them are the same ABI size.

This is easy for pointers and classes, because they will all be pointer-sized.

But in this case you have a 1-byte and a 4-byte variant, which are not compatible.

Reply:
So that was a bug we had previously with return types for FFI?

Reply:
definitely something we need to make sure we have tests for then. and we should update the release notes and tutorial to explain the limitation.

thanks for catching that. it makes sense as a limitation but i wasn't thinking of it when i reviewed.

Reply:
After discussing this with @jemc, we agreed that this check is "good enough" for our purposes, as it will catch multiple calls to the same FFI function with incompatible return types.

It's not feasible to perform the ABI size check against the return type as specified in the FFI declaration without rearchitecting how we perform the code generation for FFI functions in a significant way.

The way that the implementation works, the compiler will treat the first FFI call to a function as "authoritative", and all future calls to this function will be checked for ABI compatibility against it. This didn't cause a problem before because type checking was making sure that the return type for calls matched the return type of the declaration.